scalding-core/src/main/scala/com/twitter/scalding/mathematics/Combinatorics.scala

```diff
@@ -172,7 +172,8 @@ object Combinatorics {
     // create as many single-column pipes as the number of weights
     val pipes = allColumns.zip(weights).map(x => {
       val (name, wt) = x
-      IterableSource((0.0 to result by wt), name).read
+      val points = Stream.iterate(0.0) { _ + wt }.takeWhile(_ <= result)
```

Comment:
the linter does not like ranges with floats. I think this is the same logic, but maybe side steps some confusing cases with float ranges (not that anyone uses that code).